### DISCUSSION TITLE
Idea: fail the integration test for RuleSet if not all rules defined in it were used

### DISCUSSION THREAD

**Discussion by keradus** [PHP-CS-Fixer/PHP-CS-Fixer#7741] [Feature ideas]
Idea: fail the integration test for RuleSet if not all rules defined in it were used

**Comment by dozer111** [comment]
cool idea actually

i have own linter, and I use it: https://github.com/dozer111/projectlinter-cli/blob/master/linter/united-core/php/application/linter_test.go

it is very helpful to ensure that all the expected rules are executed

**Comment by Wirone** [comment]
@keradus if I understand the idea correctly, you would like to ensure that integration test for specific rule set contains all the require snippets that trigger each of the included rule? What about rules inherited from other rulesets included in the tested rule set? Probably #7264 can help here, so we can require only snippets for rules defined directly in the tested ruleset 🙂.
